standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/tools/schematool/commandparser/AbstractCommandParser.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore.tools.schematool.commandparser;
+
+import com.google.common.collect.Lists;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.IllegalFormatException;
+import java.util.List;
+
+/**
+ * Base implementation of NestedScriptParser abstractCommandParser.
+ */
+abstract class AbstractCommandParser implements NestedScriptParser {
+
+  private List<String> dbOpts;
+  // Depending on whether we are using beeline or sqlline the line endings have to be handled
+  // differently.
+  private final boolean usingSqlLine;
+
+  public AbstractCommandParser(String dbOpts, boolean usingSqlLine) {
+    setDbOpts(dbOpts);
+    this.usingSqlLine = usingSqlLine;
+  }
+
+  @Override
+  public boolean isPartialCommand(String dbCommand) throws IllegalArgumentException{
+    if (dbCommand == null || dbCommand.isEmpty()) {
+      throw new IllegalArgumentException("invalid command line " + dbCommand);
+    }
+    dbCommand = dbCommand.trim();
+    if (dbCommand.endsWith(getDelimiter()) || isNonExecCommand(dbCommand)) {
+      return false;
+    } else {
+      return true;
+    }
```

Comment:
This can be simplified to - 
return !(dbCommand.endsWith(getDelimiter()) || isNonExecCommand(dbCommand));